analyses/cms-open-data-ttbar/statistical_interference/drawer.py

```diff
@@ -0,0 +1,905 @@
+                0,
+                0,
+                0.5,
+                divide_value,
+            )
+            pad1_bottom.Draw()
+
+            pad2_upper = ROOT.TPad(
+                "pad2_upper" + str(channel.GetName()),
+                "pad2_upper" + str(channel.GetName()),
+                0.5,
+                divide_value,
+                1,
+                1,
+            )
+            pad2_upper.Draw()
+            pad2_bottom = ROOT.TPad(
+                "pad2_bottom" + str(channel.GetName()),
+                "pad2_bottom" + str(channel.GetName()),
+                0.5,
+                0,
+                1,
+                divide_value,
+            )
+            pad2_bottom.Draw()
+            pad1_upper.cd()
+
+            prefit_yields, prefit_unc, prefit_sample_values = self.get_yields(
+                obs_var, observables, channel_pdf, result, prefit=True
+            )  # get prefit uncert for histogram
+
+            self.hs_stacks += [
+                ROOT.THStack(
+                    "hs" + str(channel.GetName()),
+                    "hs" + str(channel.GetName()),
+                )
+            ]
+            sample_histograms = []
+
+            original_sample_bin_values = [
+                0
+            ] * channel.GetData().GetHisto().GetNbinsX()
+
+            for i, sample in enumerate(
+                channel.GetSamples()
+            ):  # loop over samples to get prefit histograms
+                hist = sample.GetHisto()
+                hist.SetFillColor(self.predefined_colors[i])
+                hist.SetLineColor(ROOT.kBlack)
+                sample_histograms += [hist]
+
+                for i in range(hist.GetNbinsX()):
+                    original_sample_bin_values[i] += hist.GetBinContent(i + 1)
+
+                self.hs_stacks[-1].Add(
+                    sample_histograms[-1]
+                )  # save histogram to THStack
+
+            channel_name = "_".join(channel.GetName().split("_")[1:])
+            self.hs_stacks[-1].SetTitle(channel_name + " PREFIT")
+            self.hs_stacks[-1].Draw("hist")
+            self.hs_stacks[-1].GetXaxis().SetLabelSize(0.06)
+
+            maximum_y_val = (
+                ROOT.gPad.GetUymax()
+            )  # used for moving upper limit of Y ax
+
+            bin_index = 1
+
+            for bin_index in range(1, sample_histograms[-1].GetNbinsX() + 1):
+                leftEdge = sample_histograms[-1].GetBinLowEdge(bin_index)
+                binWidth = sample_histograms[-1].GetBinWidth(bin_index)
+                rightEdge = leftEdge + binWidth
+
+                central_value = original_sample_bin_values[bin_index - 1]
+                unc = prefit_unc[bin_index - 1]
+                down_value = central_value - unc
+                up_value = central_value + unc
+
+                if up_value > maximum_y_val:
+                    maximum_y_val = up_value  # if box are over upper limit -> we are goint to move upper limit after
```

Comment:
```suggestion
                 if box are over upper limit -> we are going to move upper limit after
                if up_value > maximum_y_val:
                    maximum_y_val = up_value
```
typo (goint -> going)